src/support/bits.h

```diff
@@ -52,6 +53,8 @@ extern template int CountTrailingZeroes(uint32_t);
 extern template int CountTrailingZeroes(uint64_t);
 extern template int CountLeadingZeroes(uint32_t);
 extern template int CountLeadingZeroes(uint64_t);
+extern template int CeilLog2(uint32_t);
+extern template int CeilLog2(uint64_t);
```

Comment:
It looks like it would be a lot simpler to just overload the CeilLog2 for uint32_t and uint64_t without using any templates. That probably applies to all these functions, but since they're already like that, this can be left for a follow up.

Reply:
It was my first attempt, but templates in C++ still a mystery to me sometimes=) I didn’t manage to make friends with this `bits.h` header:
```cpp
template<typename T>
int CeilLog2(T v) {
  return sizeof(T) * 8 - CountLeadingZeroes(v - 1);
}
```
Also it quite hard restrict `T` to only unsigned integer types.

Reply:
Yeah, no problem, let's leave cleaning that up to a follow-up PR.